### ISSUE TITLE
Flush should probably not remove (infinitely) looping transforms

### ISSUE THREAD

**Issue by peppy** [ppy/osu-framework#900]
Right now, running `Flush` will play out all `Transforms`s to their end, ignoring their loop count. While this likely makes sense for `Transform`s with finite loop counts, it doesn't for animations you expect to never end.

A naive approach would be to ignore transforms which `HasNextIteration`, but this may not be what we want.

Consider that `Flush` is commonly used to "play out" an animation (sequence of `Transform`s) to its end. Therefore, I believe it may be beneficial to play out infinite loops to the duration of the finite animation that was flushed. Consider a rotating background that is part of a larger hierarchy, where you want the rotation to "skip" to the point it would be expected to be at when the rest of the flush occurs.

**Comment by Tom94** [comment]
I agree it's non-obvious what should happen with infinite loops. As you said, leaving them alone seems more weird since the expectation is that no transforms are funning post-Flush.

**Comment by peppy** [comment]
I see `Flush` as being "flush to a point in time" so i think any infinite loops should still be present. Every usage scenario I can think of would want this.

`ClearTransforms` on the other hand should definitely remove them.

**Comment by Tom94** [comment]
I'd prefer calling `Flush` something like `FastForward` or `Skip` in this case.

**Comment by Tom94** [comment]
In fact, implementing it as skipping a certain amount of time (double.MaxValue if we want to fully flush) would be incredibly simple and elegant.

**Comment by peppy** [comment]
Works for me

**Comment by Tom94** [comment]
I'll incorporate it into my transforms rework, since loop logic will no longer be managed by `Transformable` there, but by a (potentially infinite) chain of transforms.

**Comment by peppy** [comment]
As a heads up, when you get to looking at the osu! side of things, you'll see a use case in `MedalOverlay` (there are two rotation loops which currently get killed by flush when they shouldn't).
